components/SignUpForm/index.js

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { autobind } from 'core-decorators';
+import AuthFields from '../AuthFields';
+import validate from '../AuthFields/index.validation';
+import connect from './index.data';
+
+const defaultFunction = function defFunc() {};
+
+class SignUpForm extends React.Component {
+  static defaultProps = {
+    createUser: defaultFunction,
+    signinDispatcher: defaultFunction,
+    signinUser: defaultFunction,
+    signinUserDispatcher: defaultFunction
+  };
+
+  static propTypes = {
+    createUser: PropTypes.func,
+    signinDispatcher: PropTypes.func,
+    signinUser: PropTypes.func,
+    signinUserDispatcher: PropTypes.func
+  };
+
+  state = {
+    errors: {},
+    serverErrors: {},
+    touched: false
+  };
+
+  getServerErrors(err) {
+    if (err.graphQLErrors) {
+      const obj = {};
+      obj.message = err.graphQLErrors[0].message;
+      this.setState({
+        serverErrors: obj
+      });
+    }
+  }
+
+  formFields = [
+    { key: 1, attr: { name: 'firstName', type: 'text', label: 'First Name' } },
+    { key: 2, attr: { name: 'lastName', type: 'text', label: 'Last Name' } },
+    { key: 3, attr: { name: 'email', type: 'email', label: 'Email' } },
+    { key: 4, attr: { name: 'password', type: 'password', label: 'Password' } }
+  ];
+
+  @autobind
+  handleTouch() {
```

Comment:
@Sly777 `@autobind` very odd solution, because we already have language syntax for implicit binding. Next.js includes all needed babel presets and this syntax works out of the box.

> But we can change it to that if there's no difference between class-y functions and variable functions inside of class.

I think that class property initialization with arrow function the most cleanest solution for binding in React components and here no difference for this case, but if look deeper, here exists inessential diff between class-y function and function with property initialization https://babeljs.io/docs/plugins/transform-class-properties